### PR TITLE
Ds 14 today news home screen

### LINKED ISSUE #14: display Today news on home screen
**Is your feature request related to a problem? Please describe.**
Populate home screen of app with an additional news section, labeled as "Today".

**Describe the solution you'd like**
New heading tag, with custom content for section.

### PR DESCRIPTION
**Pull request by dannysanchez559**
## Changes
_List Changes Introduced by this PR_
1. Add interface implementation for TodayNews endpoint
2. Add TodayNews component
3. Add styled-components file for TodayNews
4. Display Featured news article with bare styling
5. Display Standard news articles with proper grouping and bare styling

Closes #14

### CODE REVIEW

File: modern-news-app/src/App.tsx [hunk 1]
```diff
@@ -1,17 +1,35 @@
...
+  }, []);
+
+  // // news data call for top section of home page
+  const getNewsData = async () => {
+=======
     getNewsData("bitcoin")
   }, [])
 
   const getNewsData = async (search: string) => {
+>>>>>>> 26b6d3eb7f70df0ce5e204d189d806c61ef842e1
```
Comment (kmausolf) [inline modern-news-app/src/App.tsx:32]: Looks like a merge artifact.  Do you need help resolving the merge?

File: modern-news-app/src/util/fetchApi.ts [hunk 2]
```diff
@@ -1,13 +1,37 @@
...
         const newsResponse = await fetch(baseUrl).then(res => res.json());
         newsResponse.search = request.q
         return newsResponse
     } catch (error) {
         console.error(error)
     }
-}
\ No newline at end of file
+}
+>>>>>>> 26b6d3eb7f70df0ce5e204d189d806c61ef842e1
```
Comment (kmausolf) [inline modern-news-app/src/util/fetchApi.ts]: Another merge artifact.  Let's clean these up before review.

**Review by kmausolf — CHANGES_REQUESTED** [review]
Looks like some merge artifacts are in the code, possibly an incomplete merge.  Let me know if you need help straightening these out.

File: modern-news-app/src/App.tsx [hunk 3]
```diff
@@ -21,11 +33,25 @@ const App = () => {
...
+      return dataObject;
+
+    } catch (error) {
+        console.error(error);
+    }
+}
 
   return (
     <div className="App" style={{ backgroundColor: "#22242C", color: "white", minHeight: "100vh"}}>
+      {console.log("newsData ", newsData)}
```
Comment (kmausolf) [inline modern-news-app/src/App.tsx:51]: Log here

Comment (Gjhnsn) [inline modern-news-app/src/App.tsx:51]: need this console.log still?

File: modern-news-app/src/App.tsx [hunk 4]
```diff
@@ -1,15 +1,27 @@
...
+import { fetchTodayNewsData } from './util/fetchApi';
+import { ITodayNewsResponse, TodayNewsRequest, TodayNewsResponse } from './models/TodayArticles';
+
+// components 
+import TodayNews from './components/TodayNews';
 import Navbar from './components/Navbar';
 import Main from './components/Main';
+
+// react
 import { useEffect, useState } from 'react';
```
Comment (krferrell) [inline modern-news-app/src/App.tsx:13]: I really like this organization of imports

**Review by kmausolf — COMMENTED** [review]
Something about the diff doesn't add up, but the PR seems fine.  Can I have someone do a test run to make sure the app loads correctly?

**Review by kmausolf — APPROVED** [review]

File: modern-news-app/src/App.tsx [hunk 5]
```diff
@@ -1,15 +1,27 @@
...
 
 const App = () => {
-  const [newsData, setNewsData] = useState<IEverythingNewsResponse>(new EverythingNewsResponse())
+  const [newsData, setNewsData] = useState<IEverythingNewsResponse>(new EverythingNewsResponse());
+  const [todayNewsData, setTodayNewsData] = useState<ITodayNewsResponse>(new TodayNewsResponse());
 
   useEffect(() => {
-    getNewsData("bitcoin")
-  }, [])
+    // getNewsData("bitcoin");
```
Comment (Lap343) [inline modern-news-app/src/App.tsx:20]: I would just get rid of this

**Review by Gjhnsn — APPROVED** [review]
Looks good, great job documenting with comments!

File: modern-news-app/src/App.tsx [hunk 6]
```diff
@@ -1,15 +1,27 @@
...
+    getTodayNewsData();
+  }, []);
+
+  // news data call for top section of home page
 
   const getNewsData = async (search: string) => {
```
Comment (Lap343) [inline modern-news-app/src/App.tsx:26]: same thing with this func since we dont need it anymore

**Review by Lap343 — APPROVED** [review]
The code looks good other than the styled components. It just looks way to bloated. I would recommend for a later refactor. Nest those styles within each other and it will look a lot cleaner.

File: modern-news-app/src/util/fetchApi.ts [hunk 7]
```diff
@@ -1,13 +1,26 @@
...
+export const fetchTodayNewsData = async (request: ITodayNewsRequest) => {
+    try {
+        // url for US news with todays date
+        const todayNewsUrl = `${baseUrl}top-headlines?country=${request.sources}&apiKey=${request.apiKey}&pageSize=${request.pageSize}&publishedAt=2021-08-25`;
+        return await fetch(todayNewsUrl).then(res => res.json());
+
+    } catch (error) {
+        console.error(error);
+    }
+}
```
Comment (krferrell) [inline modern-news-app/src/util/fetchApi.ts:26]: Might be a good idea to use an interface here for the response to make sure you are getting back what you are looking for.

**Review by krferrell — APPROVED** [review]
Nice work. Everything looking solid.

**Pull request merged**